### ISSUE TITLE
Show a reminder dialog before being redirected to external URL

### ISSUE THREAD

**Issue by think-in-universe** [near/near-wallet#1963]
## Story

As a wallet user who want to avoid being phished by 3rd party web applications, I want to be reminded before I was redirected to an external link, so I can double check the link before I confirm the redirect.

For example, in the case of `redirectUrl` of linkdrop, the wallet may show a popup window first before being redirected to the `redirectUrl`.

## Acceptance Criteria

- [ ] Before being redirected to an external URL, show a popup dialog that explicilty saying that you'll be redirected to a URL 
- [ ] The full path of the `redirectUrl` will be shown in the dialog to ensure user reviewed the URL before move forward
- [ ] Mobile ok
- [ ] Tested

## NOTES

- This comes from discussion with developers who has used linkdrop redirectUrl intenstively for their project.

**Comment by stefanopepe** [comment]
Adding @corwinharrell since we may need some UX prototyping.
I give green light to the feature, it's overdue and would benefit all user base

**Comment by corwinharrell** [comment]
In cases where we do need to show a dialogue, and cannot reference the redirect inline, we can use our dialogue pattern to prompt our user before the redirect occurs.

**Implementation Details**

- The redirect should occur automatically within a time limit (10sec in the example below).
- The text that indicates the time limit should be dynamic, and should count down in real time.
- Alternatively, the user may accelerate the redirect by clicking `Continue to Site`.
- By clicking `Cancel` the user would simply be taken back to the previous step in whatever flow it is they are completing.

<img width="1318" alt="Screen Shot 2022-01-31 at 10 45 57 AM" src="https://user-images.githubusercontent.com/5934188/151826199-e54083c2-0723-4123-a366-af3f34df4188.png">

**Comment by corwinharrell** [comment]
@think-in-universe @stefanopepe In the case of something like connecting to a dApp, it would make the most sense to just include a message inline at the final confirmation/confirm step, saying "Once confirmed, you will be redirected" rather than displaying a dialogue when it isn't necessary.

We'll need to identify these instances so we can determine how best to include the messaging inline. Not every redirect should use the dialogue solution shared above.

**Comment by corwinharrell** [comment]
> It'd be helpful to have it explicitly specified i.e what flows use the dialog and which ones do not?

yep agreed @gutsyphilip will need to lean on others from engineering to identify where each of these instances might be. cc @MaximusHaximus @Patrick1904

**Comment by think-in-universe** [comment]
@corwinharrell the most common case I have in mind is linkdrop, when user received a linkdrop with a `redirectUrl`, she/he might be redirected to a phishing page (e.g. a fake web wallet which is not easy to realize sometimes in mobile phone).

I agree that the extra dialog will hurt the UX especially when the redirect actually makes sense (e.g. claim a NFT in the redirected page), but it would be hard to differentiate the good and malicious `redirectUrl`.

**Comment by Patrick1904** [comment]
> > It'd be helpful to have it explicitly specified i.e what flows use the dialog and which ones do not?
> 
> yep agreed @gutsyphilip will need to lean on others from engineering to identify where each of these instances might be. cc @MaximusHaximus @Patrick1904

@gutsyphilip the best indicator for this is the `redirectToApp` as it's the most commonly used function for redirecting to a page site outside the wallet.

@corwinharrell is the idea that we're going to show this redirect modal before sending users back to the app after signing in or signing an app tx too? Or is this mainly for flows where it may be more of an unexpected path, like claiming a linkdrop?

**Comment by corwinharrell** [comment]
> Is the idea that we're going to show this redirect modal before sending users back to the app after signing in or signing an app tx too?

As mentioned above I think displaying this dialogue _every_ time there is a redirect is very cumbersome and would be a very suboptimal UX. I agree with @stefanopepe that if we are redirecting the user _back_ to a site that they have just come from (for txn confirmation for example) we don't need to display this dialogue. This dialogue should be reserved for instances where there is a greater phishing risk or if the referrer is unknown.

> I think an audit may be needed first to decide exactly when to and when not to show the modal.

Agreed. I think the engineering team is best positioned to identify where these instances all occur. @Patrick1904 would you and Philip be able to construct a list of redirect instances? Design for this issue is unlikely to change all that much, it's mostly a matter of deciding when and when not to display the dialogue.
